app/src/main/java/org/p2p/wallet/auth/common/WalletAuthManager.kt

```diff
@@ -0,0 +1,120 @@
+package org.p2p.wallet.auth.common
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.os.Build
+import android.webkit.JavascriptInterface
+import android.webkit.WebView
+import android.widget.Toast
+import com.google.gson.Gson
+import org.p2p.wallet.auth.model.DeviceShareKey
+import org.p2p.wallet.auth.model.GoogleAuthFlow
+import org.p2p.wallet.infrastructure.security.SecureStorageContract
+
+private const val KEY_DEVICE_SHARE = "KEY_DEVICE_SHARE"
+private const val indexUri = "file:///android_asset/index.html"
+private const val communicationChannel = "AndroidCommunicationChannel"
+
+class WalletAuthManager(
```

Comment:
you can put some impl details. how this auth works
smth like
WalletWeb3AuthManager